internal/kldcontracts/smartcontractgw.go

```diff
@@ -1146,12 +1147,30 @@ func (g *smartContractGW) addABI(res http.ResponseWriter, req *http.Request, par
 		return
 	}
 
-	preCompiled, err := g.compileMultipartFormSolidity(tempdir, req)
+	abi, err := g.parseABI(req.Form)
 	if err != nil {
-		g.gatewayErrReply(res, req, klderrors.Errorf(klderrors.RESTGatewayCompileContractCompileFailed, err), 400)
+		// TODO: Find appropriate error message
```

Comment:
The `// TODO` should have been removed. Thanks for catching that.

Currently the code returns `klderrors.RESTGatewayCompileContractInvalidFormData` because this error happens when it finds something in the `abi` field, but can't unmarshal it as an ABI. This seems to be the error we normally return when we can't parse a field, but I could add ones for each specific field if we want to provide that level of detail.

Reply:
So you'll remove the `TODO`?